test/lib/ufe/testRename.py

```diff
@@ -267,3 +267,39 @@ def testRenameRestrictionOtherLayerOpinions(self):
         with self.assertRaises(RuntimeError):
            newName = 'Ball_35_Renamed'
            cmds.rename(newName)
+
+    def testRenameRestrictionExternalReference(self):
+        # open tree_ref.ma scene in test-samples
+        mayaUtils.openTreeRefScene()
+
+        # clear selection to start off
+        cmds.select(clear=True)
+
+        # select a USD object.
+        mayaPathSegment = mayaUtils.createUfePathSegment('|world|TreeRef_usd|TreeRef_usdShape')
+        usdPathSegment = usdUtils.createUfePathSegment('/TreeBase/leaf_ref_2')
+        treebasePath = ufe.Path([mayaPathSegment, usdPathSegment])
+        treebaseItem = ufe.Hierarchy.createItem(treebasePath)
+
+        ufe.GlobalSelection.get().append(treebaseItem)
+
+        # get the USD stage
+        stage = mayaUsd.ufe.getStage(str(mayaPathSegment))
+
+        # set the edit target to the root layer
+        stage.SetEditTarget(stage.GetRootLayer())
+        self.assertEqual(stage.GetEditTarget().GetLayer(), stage.GetRootLayer())
+        self.assertTrue(stage.GetRootLayer().GetPrimAtPath("/TreeBase"))
+
+        # create a leafRefPrim
+        leafRefPrim = usdUtils.getPrimFromSceneItem(treebaseItem)
+
+        # leafRefPrim should have an Authored References
+        self.assertTrue(leafRefPrim.HasAuthoredReferences())
+
+        # expect the exception to happen
+        with self.assertRaises(RuntimeError):
+           newName = 'leaf_ref_2_renaming'
+           cmds.rename(newName)
+
```

Comment:
Added a simple unit test around rename restriction for an external reference.

NOTE: I originally didn't want to add `treeRef.usda` and `treeRef.ma` and my plan was to use already added `tree.ma` and programmatically add internal reference, however AddInternalReference didn't seem to give me what I wanted.

What I wanted was to add a `typeless` prim, append an internal reference, author bunch of attributes and be done.
e.g 
```
    def "leaf_ref_1" (
        append references = </TreeBase/leavesXform/greenLeaf>
    )
    {
        double3 xformOp:translate = (2.0, 2.5, 0)
        uniform token[] xformOpOrder = ["xformOp:translate"]
        double radius = 0.5
    }
```

Here is how I originally planned to add in code:
```
refPrim = stage.DefinePrim('/TreeBase/leaf_ref_1')
refPrim.GetReferences().AddInternalReference('/TreeBase/leavesXform/greenLeaf')

# author refPrim's xform and radius
xform = UsdGeom.Xformable(refPrim)
xform.AddTranslateOp().Set(Gf.Vec3d(2.0, 2.5, 0))

radiusAttr = refPrim.GetAttribute('radius')
radiusAttr.Set(0.5)
```
Instead I get something like this:
```
def Sphere "leaf_ref_1"
{
    float3[] extent = [(-2, -2, -2), (2, 2, 2)]
    color3f[] primvars:displayColor = [(0, 1, 0)]
    double radius = 0.5
    double3 xformOp:translate = (2, 2.5, 0)
    uniform token[] xformOpOrder = ["xformOp:translate"]
}
```